pkg/apis/kubermatic/v1/policy_template.go

```diff
@@ -44,22 +43,6 @@ type PolicyTemplate struct {
 }
 
 type PolicyTemplateSpec struct {
-	// Title is the title of the policy, specified as an annotation in the Kyverno policy
-	Title string `json:"title"`
-
-	// Description is the description of the policy, specified as an annotation in the Kyverno policy
-	Description string `json:"description"`
-
-	// Category is the category of the policy, specified as an annotation in the Kyverno policy
-	//
-	// +optional
-	Category string `json:"category,omitempty"`
-
-	// Severity indicates the severity level of the policy
-	//
-	// +optional
-	Severity string `json:"severity,omitempty"`
-
```

Comment:
Why are we removing these fields?

Reply:
We will be passing the resource as a Kubernetes resource, so it will have the apiVersion, kind, metadata and spec. These fields were to be passed as annotations in the Kyverno policy (the first three fields), now they can be passed directly and validated from the webhook.

Reply:
> We will be passing the resource as a Kubernetes resource, so it will have the apiVersion, kind, metadata and spec

Why would we embed the complete CR within the PolicyTemplate CR when we only just need the spec?

Reply:
> These fields were to be passed as annotations in the Kyverno policy (the first three fields), now they can be passed directly and validated from the webhook.

This would make things even tougher to deal with since for example, title and description are mandatory fields. 

1. We'll have to rely on type casting and some webhooks in the code that will verify if they are provided.
2. Instead of using spec, API will also have to first typecast, retrieve object, go through annotations and then find these fields.

Reply:
I'm also torn between embedding entire objects or just spec. My thought was that if we use x-kubernetes-embed-resource (or similar), we could technically support different policy engines, because we'd have apiVersion/kind available.

But getting rid of all the fields here might be a step to far, as I can see them be useful in KKP (like the severity, which could be a additional kubectl printer column). So without knowing much more, I would agree with Waleed and rather keep these fields in our CRD.

Reply:
The idea for having it like that was to make it extensible in the future, we could also support different policy frameworks. Considering the dashboard needs these fields I can add them back and revert to a Spec field like it was, but will only change the name to something more generic.